test/check-machines-multi-host-consoles

```diff
@@ -0,0 +1,135 @@
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import testlib
+from machineslib import VirtualMachinesCase
+
+
+def start_machine_troubleshoot(b, new=False, known_host=False, password=None):
+    b.wait_visible("#machine-troubleshoot")
+    b.click('#machine-troubleshoot')
+    b.wait_visible('#hosts_setup_server_dialog')
+    if new:
+        b.click('#hosts_setup_server_dialog button:contains(Add)')
+        if not known_host:
+            b.wait_in_text('#hosts_setup_server_dialog', "You are connecting to")
+            b.wait_in_text('#hosts_setup_server_dialog', "for the first time.")
+            b.click("#hosts_setup_server_dialog button:contains('Trust and add host')")
+    if password:
+        b.wait_in_text('#hosts_setup_server_dialog', "Unable to log in")
+        b.set_input_text('#login-custom-password', password)
+        b.click('#hosts_setup_server_dialog button:contains(Log in)')
+
+
+def add_machine(b, address, known_host=False, password="foobar"):
+    b.switch_to_top()
+    b.go(f"/@{address}")
+    start_machine_troubleshoot(b, new=True, known_host=known_host, password=password)
+    b.wait_not_present('#hosts_setup_server_dialog')
+    b.enter_page("/system", host=address)
+
+
+class TestMultiMachineVNC(VirtualMachinesCase):
+    """
+    Test for showing the wrong VNC console when adding another host via the add
+    host feature the raw stream channel does not set the host option so would
+    connect to the console of machine you connect to instead of the newly added
+    host.
+
+    https://issues.redhat.com/browse/RHEL-3959
+    """
+
+    provision = {  # noqa: RUF012
+        "machine1": {"address": "10.111.113.1/20", "memory_mb": 660},
+        "machine2": {"address": "10.111.113.2/20", "memory_mb": 660},
+    }
+
+    def setup_ssh_auth(self):
+        self.machine.execute("d=/home/admin/.ssh; mkdir -p $d; chown admin:admin $d; chmod 700 $d")
+        self.machine.execute("test -f /home/admin/.ssh/id_rsa || ssh-keygen -f /home/admin/.ssh/id_rsa -t rsa -N ''")
+        self.machine.execute("chown admin:admin /home/admin/.ssh/id_rsa*")
+        pubkey = self.machine.execute("cat /home/admin/.ssh/id_rsa.pub")
+
+        for m in self.machines:
+            self.machines[m].execute("d=/home/admin/.ssh; mkdir -p $d; chown admin:admin $d; chmod 700 $d")
+            self.machines[m].write("/home/admin/.ssh/authorized_keys", pubkey)
+            self.machines[m].execute("chown admin:admin /home/admin/.ssh/authorized_keys")
+
+    def setUp(self):
+        super().setUp()
+        self.machine1 = self.machines['machine1']
+        self.machine2 = self.machines['machine2']
+
+        # Disable preloading on all machines
+        # Preloading on machines debug build can overload the browser and cause slowness and browser crashes,
+        # and failing to load sofware updates breaks pixel tests in release builds
+        self.setup_provisioned_hosts(disable_preload=True)
```

Comment:
Please check if it's really necessary. If so, please adjust the  comment, as that's a copy pasta. If not, please drop.